### DISCUSSION TITLE
Fortran runtime error: Array bound mismatch for dimension 3 of array 'gwann' (601/15)

### DISCUSSION THREAD

**Discussion by wanyuloo** [nakib/elphbolt#155] [General]
Description:
While attempting to compute gk_qpath, the calculation was interrupted due to a Fortran runtime error related to array bounds mismatch. Specifically, the error message indicates an issue with the third dimension of the array gwann.
Error Message:
Fortran runtime error: Array bound mismatch for dimension 3 of array 'gwann' (601/15)
Context:
I am running the gk_qpath calculation, and the error occurs during the process. Based on the error message, it appears that the array gwann is expected to have a size of 15 for the third dimension, but the actual size encountered during the calculation is 601, causing a mismatch and interrupting the execution.
The issue arises consistently and halts the computation.
Steps Taken:
I have provided highsympath.txt and initialk.txt files as input, and the calculation proceeds up until this error occurs.
I have verified the input files and the initial setup, but the issue persists.

**Comment by nakib** [comment]
Dear wanyloo,

Thanks for writing. I have not encountered this issue before. I will need to see your output file to understand what is happening.

Best,
Nakib

**Comment by wanyuloo** [comment]
In my previous inquiry, I encountered a Fortran runtime error when calculating a specific material using elphbolt. To further troubleshoot, I decided to run the official silicon example from the elphbolt website. Unfortunately, the same issue occurred.
The error message and output file are as follows：
[error message.log](https://github.com/user-attachments/files/17258092/error.message.log)
[output.log](https://github.com/user-attachments/files/17258094/output.log)

**Comment by nakib** [comment]
Thanks. Your output shows several instances of `Number of coarray images =     1`, from which it seems to me that your coarray installation is faulty. The quantity `gwann` is distributed over several coarray images and without a proper installation of coarrays, this error is inevitable.

I am moving this to the discussion section.

Best,
Nakib

**Comment by wanyuloo** [comment]
Could you please provide guidance on how to correctly install and configure coarrays to ensure the proper execution of elphbolt?

**Reply by dpaulzc** [reply]
Hi,

I can help regarding this issue. 

First check if mpich is preinstalled in your system. If it is already installed, and it is not built during the installation of open coarrays, then you have to uninstall it. To install opencoarrays, use ` ./install.sh --with-fortran <path to>/gfortran-12 --with-cxx <path to>/g++-12 --with-c <path to>/gcc-12 `. Use version 12 of gcc, g++ and gfortran. There seems to be some compatibility issues with version 13. Add the paths to caf and cafrun executable to the $PATH (The paths to the exec are printed during the installation). Let me know if you have any queries.

**Reply by nakib** [reply]
Hi Dwaipayan,

Why should `./install.sh --with-fortran /gfortran-12 --with-cxx /g++-12 --with-c /gcc-12 ` work? There is typically no binary called `gfortran-12`, etc. in the `/` directory of any unix system. Did you mean to write `<path to>/gcc-12`? The `<path to>` can be found by writing `which gcc-12`. And similar for `g++-12`, `gfortran-12`, etc.

Best,
Nakib

**Reply by dpaulzc** [reply]
Yes, you are correct. I wrote the `<path>` , but it didn't appear in the comment.
